platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/AttributeContainerResolutionIntegrationTest.groovy

```diff
@@ -42,8 +42,10 @@ class AttributeContainerResolutionIntegrationTest extends AbstractIntegrationSpe
                 inputs.files(configurations.conf)
             }
         """
+
         expect:
         // When this has failed in the past, building the task graph hits a NPE from the attribute container
+        executer.expectDocumentedDeprecationWarning("Querying the contents of an attribute container while realizing attributes of the container. This behavior has been deprecated. This will fail with an error in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/current/userguide/upgrading_version_8.html#attribute_container_recursive_query")
```

Comment:
💭 When someone sees this message, the immediate next question would be: Which container is being queried?

Can we do anything to provide information here that would help them identify where changes need to be made?  Even attribute keys being queried might help.

Reply:
The AttributeContainer neither has a name or knows its owner. 

However, I don't think this will be much of an issue. First, I think this is quite a rare case to run into this deprecation and second, the stack trace should show you exactly where this deprecation occurs.

Reply:
If only the stacktrace suggestion worked: https://github.com/gradle/gradle/pull/31821.  I guess you always get the first line, which is good.

Still, I think there's an argument to be had that if you are a thing-with-attributes, you are important enough to always have a name.  I wonder how tricky it would be to add a `getDisplayName()` to `HasAttributes`, or make it extend `Describable` and record names in the constructors of all the implementations that don't already do this?  Something to try and see, I guess.

Reply:
I wouldn't add onto `HasAttributes` at all -- I think we should remove that interface entirely.

I don't necessarily think we need to modify the interface of the `AttributeContainer` as well -- we could pass in a `Describable` of its owner into the constructor and leverage that owner description in this message. But all that extra wiring to pass in the owner increases complexity and maintenance burden, and I'm not so sure its worth it here.

In any case totally agreed we should fix that incorrect suggestion for the deprecation warnings. It always gets me and I have to look up the proper flag to set